footylib/footylib.py

```diff
@@ -197,71 +202,59 @@ def _populate(self, info, division):
-        # TODO: look for NL locale
-        english_datetime = months[dutch_datetime[0]]
-        dutch_datetime[0] = english_datetime
-        english_datetime = " ".join(dutch_datetime)
+        locale.setlocale(locale.LC_TIME, 'nl_NL')
+        dutch_datetime = '{} {}'.format(date.capitalize(), time)
         try:
-            datetime_object = datetime.strptime(english_datetime,
+            datetime_object = datetime.strptime(dutch_datetime,
                                                 '%B %d, %Y %I:%M %p')
             return datetime_object
         except ValueError:
             LOGGER.exception("Couldn't parse this datetime.")
 
 
 class FootyCalendar(object):
-    def __init__(self, season):
+    def __init__(self, datetime, teams, location):
         self.logger = logging.getLogger('{base}.{suffix}'.format(
             base=LOGGER_BASENAME, suffix=self.__class__.__name__))
         self.timezone = 'Europe/Amsterdam'
         self.calendar = Calendar()
-        self.season = season
+        self.datetime = datetime
+        self.teams = teams
+        self.location = location
 
     def generate(self):
-        for match in self.season:
-            event = Event(duration=timedelta(hours=1))
-            event.begin = pytz.timezone(self.timezone).localize(
-                                                match.get('time'))
-            try:
-                # ics module escapes strings. Needs decoding.
-                event.name = match.get('match').decode('utf-8')
-            except (UnicodeEncodeError, UnicodeDecodeError):
-                event.name = match.get('match')
-                self.logger.exception('Unicode Error. '
-                                      'Got {event} {type}'.format(
-                                                          event=event.name,
-                                                          type=type(event.name)))
-            except AttributeError:
-                self.logger.exception("Got an exception in calendar")
-
-            self.calendar.events.append(event)
+        event = Event(duration=timedelta(hours=1))
+        try:
+            event.begin = pytz.timezone(self.timezone).localize(self.datetime)
+        except AttributeError:
+            self.logger.exception('{} not valid datetime'.format(self.datetime))
+        try:
+            # ics module escapes strings. Needs decoding.
+            event.name = self.teams.decode('utf-8')
+        except (UnicodeEncodeError, UnicodeDecodeError):
+            self.logger.exception('Unicode Error. '
+                                  'Got {event} {type}'.format(
+                                                      event=event.name,
+                                                      type=type(event.name)))
+            event.name = self.teams
+        except AttributeError:
+            self.logger.exception("Got an exception in calendar")
+
+        self.calendar.events.append(event)
```

Comment:
Here is where you append the event to the calendar. Now, what if instead of adding the event to a calendar, you were just returning a list of events and you constructed the actual calendar object right before you return on each instance of the calendar attribute?